src/app/shared-services/inferences/engine/staypoint-engine/index.ts

```diff
@@ -0,0 +1,106 @@
+import { Inference } from 'src/app/model/inference'
+import { Trajectory } from 'src/app/model/trajectory'
+import {
+  IInferenceEngine,
+  InferenceDefinition,
+  InferenceResult,
+  InferenceResultStatus,
+  InferenceType,
+} from '../types'
+import { NightnessScoring } from '../../scoring/nightness-scoring'
+import { IInferenceScoring } from '../../scoring/types'
+import { WorkHoursScoring } from '../../scoring/work-hours-scoring'
+import {
+  inferHomeFromStayPointClusters,
+  inferWorkFromStayPointClusters,
+} from 'src/app/shared-services/staypoint/utils'
+import { StaypointService } from 'src/app/shared-services/staypoint/staypoint.service'
+
+export class StaypointEngine implements IInferenceEngine {
+  scorings: IInferenceScoring[] = [
+    new NightnessScoring(),
+    new WorkHoursScoring(),
+  ]
+
+  constructor(private staypointService: StaypointService) {}
+
+  private inputCoordinatesLimit = 100000
+
+  async infer(
+    trajectory: Trajectory,
+    inferences: InferenceDefinition[]
+  ): Promise<InferenceResult> {
+    if (trajectory.coordinates.length > this.inputCoordinatesLimit) {
```

Comment:
Just a quick note and possibly not highly relevant:
This limitation was necessary with the old `simple-engine`, because we're running DBScan for _all_ locations. In the case of this `staypoint-engine` this _might_ be obsolet (or at least a higher limit could be viable).

Reply:
This is a good point and I agree it might be obsolete.

I'd not change it for now due to the issue with the test trajectory we talked about once: the nepal test trajectory is not saved in the database and hence there are issues with ID as the foreign key when trying to save its inferences to the database. I believe the simple-engine is only avoiding running into this issue through this check, so I am leaving it in here for now as well. At some point it would probably make sense to address this edge case with the test trajectory type, I suggest removing this check after.